src/Apps/Order/Routes/Payment/index.tsx

```diff
@@ -435,7 +406,9 @@ export class PaymentRoute extends Component<PaymentProps, PaymentState> {
 }
 
 export const PaymentFragmentContainer = createFragmentContainer(
-  injectStripe(trackPageViewWrapper(injectDialog(PaymentRoute))),
+  injectCommitMutation(
+    injectStripe(trackPageViewWrapper(injectDialog(PaymentRoute)))
+  ),
```

Comment:
Or something like

```js
const decorate = compose(
  injectCommitMutation, 
  injectStripe, 
  injectDialog, 
  trackPageViewWrapper
) 

export const PaymentFragmentContainer = createFragmentContainer(decorate(PaymentRoute), ...)
```

Reply:
Ideally we could do 

```js
@injectCommitMutation
@injectStripe
@injectDialog
@trackPageViewWrapper
export class Foo extends Component { ... }
```

Which a few different devs and I have looked at in the past, but couldn't figure out the type properly, and so we tabled it. Since we already use the `@track` decorator all over, and top-level decorators can be simple HOCs, it makes sense to keep these things together if possible. But also don't want to increase complexity by introducing intermediary layers _just_ to decorate since wrapping is the same thing. 

:silverbullet:

Reply:
whoaaaa